tests/test_elasticsearch_eager.py

```diff
@@ -0,0 +1,127 @@
+"""Tests for the elasticsearch datasets"""
+
+import time
+import json
+import pytest
+import docker
+from docker.errors import NotFound
+import requests
+import tensorflow as tf
+import tensorflow_io as tfio
+
+# COMMON VARIABLES
+
+ES_CONTAINER_NAME = "tfio-elasticsearch"
+NODE = "http://localhost:9200"
+INDEX = "people"
+DOC_TYPE = "survivors"
+HEADERS = {"Content-Type": "application/json"}
+
+
+def is_container_running():
+    """Check whether the elasticsearh container is up and running"""
+
+    client = docker.from_env()
+    try:
+        es_container = client.containers.get(ES_CONTAINER_NAME)
+        if es_container.status == "running":
+            return True
+        return False
+    except NotFound:
+        return False
+
+
+@pytest.mark.skipif(not is_container_running(), reason="The container is not running")
+def test_create_index():
+    """Create an index in the cluster"""
+
+    create_index_url = "{}/{}".format(NODE, INDEX)
+    res = requests.put(create_index_url)
+    assert res.status_code == 200
+
+
+@pytest.mark.parametrize(
+    "name,gender,age,fare,survived",
+    [
+        ("person1", "Male", 20, 80.52, 1),
+        ("person2", "Female", 30, 40.88, 0),
+        ("person3", "Male", 40, 20.73, 0),
+        ("person4", "Female", 50, 100.99, 1),
+    ],
+)
+@pytest.mark.skipif(not is_container_running(), reason="The container is not running")
+def test_populate_data(name, gender, age, fare, survived):
+    """Populate the index with data"""
+
+    put_item_url = "{}/{}/{}".format(NODE, INDEX, DOC_TYPE)
+    data = {
+        "name": name,
+        "gender": gender,
+        "age": age,
+        "fare": fare,
+        "survived": survived,
+    }
+
+    res = requests.post(put_item_url, json=data, headers=HEADERS)
+
+    # The 201 status code indicates the documents have been properly indexed
+    assert res.status_code == 201
+
+    # allow the cluster to index in the background.
+    time.sleep(1)
+
+
+@pytest.mark.skipif(not is_container_running(), reason="The container is not running")
+def test_elasticsearch_io_dataset():
+    """Test the functionality of the ElasticsearchIODataset"""
+
+    dataset = tfio.experimental.elasticsearch.ElasticsearchIODataset(
+        nodes=[NODE], index=INDEX, doc_type=DOC_TYPE
+    )
+
+    assert issubclass(type(dataset), tf.data.Dataset)
+
+    for item in dataset:
+        for attr in ["name", "gender", "age", "fare", "survived"]:
+            assert attr in item
+
+
+@pytest.mark.skipif(not is_container_running(), reason="The container is not running")
+def test_elasticsearch_io_dataset_batch():
+    """Test the functionality of the ElasticsearchIODataset"""
+
+    BATCH_SIZE = 2
+    dataset = tfio.experimental.elasticsearch.ElasticsearchIODataset(
+        nodes=[NODE], index=INDEX, doc_type=DOC_TYPE
+    ).batch(BATCH_SIZE)
+
+    assert issubclass(type(dataset), tf.data.Dataset)
+
+    for item in dataset:
+        for attr in ["name", "gender", "age", "fare", "survived"]:
```

Comment:
Move this list to a constant since it appears in multiple places

Reply:
addressed in the commit